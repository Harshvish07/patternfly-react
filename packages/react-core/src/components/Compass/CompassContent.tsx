import { Drawer, DrawerContent, DrawerProps } from '../Drawer';
import styles from '@patternfly/react-styles/css/components/Compass/compass';
import { css } from '@patternfly/react-styles';

interface CompassContentProps extends React.HTMLProps<HTMLDivElement> {
  /** Content of the main compass area. Typically one or more CompassPanel components. */
  children: React.ReactNode;
  /** Additional classes added to the CompassContent */
  className?: string;
  /** Content rendered in an optional drawer wrapping the CompassContent */
  drawerContent?: React.ReactNode;
  /** Additional props passed to the drawer */
  drawerProps?: DrawerProps;
}

export const CompassContent: React.FunctionComponent<CompassContentProps> = ({
  children,
  className,
  drawerProps,
  drawerContent,
  ...props
}) => {
  const hasDrawer = drawerContent !== undefined;

  const compassContent = (
    <div className={css(styles.compassContent, className)} {...props}>
      {children}
    </div>
  );

  if (hasDrawer) {
    return (
      <Drawer {...drawerProps}>
        <DrawerContent panelContent={drawerContent}>{compassContent}</DrawerContent>
      </Drawer>
    );
  }

  return compassContent;
};

CompassContent.displayName = 'CompassContent';
