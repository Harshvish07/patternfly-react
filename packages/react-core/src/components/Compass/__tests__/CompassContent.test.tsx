import { render, screen } from '@testing-library/react';
import { CompassContent } from '../CompassContent';
import styles from '@patternfly/react-styles/css/components/Compass/compass';

test('Renders with children', () => {
  render(<CompassContent>Test content</CompassContent>);
  expect(screen.getByText('Test content')).toBeVisible();
});

test('Renders with custom class name when className prop is provided', () => {
  render(<CompassContent className="custom-class">Test</CompassContent>);
  expect(screen.getByText('Test')).toHaveClass('custom-class');
});

test(`Renders with default ${styles.compassContent} class`, () => {
  render(<CompassContent>Test</CompassContent>);
  expect(screen.getByText('Test')).toHaveClass(styles.compassContent);
});

test('Renders with drawer when drawerContent is provided', () => {
  render(<CompassContent drawerContent={<div>Drawer content</div>}>Test</CompassContent>);
  expect(screen.getByText('Drawer content')).toBeVisible();
});

test('Renders with additional props spread to the component', () => {
  render(<CompassContent aria-label="Test label">Test</CompassContent>);
  expect(screen.getByText('Test')).toHaveAccessibleName('Test label');
});

test('Matches the snapshot without drawer', () => {
  const { asFragment } = render(
    <CompassContent>
      <div>Test content</div>
    </CompassContent>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('Matches the snapshot with drawer', () => {
  const { asFragment } = render(
    <CompassContent drawerContent={<div>Drawer content</div>} drawerProps={{ isExpanded: true }}>
      <div>Test content</div>
    </CompassContent>
  );
  expect(asFragment()).toMatchSnapshot();
});
