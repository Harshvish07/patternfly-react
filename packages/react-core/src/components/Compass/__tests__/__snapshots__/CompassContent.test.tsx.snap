// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot with drawer 1`] = `
<DocumentFragment>
  <div
    class="pf-v6-c-drawer pf-m-expanded"
  >
    <div
      class="pf-v6-c-drawer__main"
    >
      <div
        class="pf-v6-c-drawer__content"
      >
        <div
          class="pf-v6-c-compass__content"
        >
          <div>
            Test content
          </div>
        </div>
      </div>
      <div>
        Drawer content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Matches the snapshot without drawer 1`] = `
<DocumentFragment>
  <div
    class="pf-v6-c-compass__content"
  >
    <div>
      Test content
    </div>
  </div>
</DocumentFragment>
`;
